src/main/java/cn/nukkit/block/BlockUndyedShulkerBox.java

```diff
@@ -112,6 +115,13 @@ public boolean place(Item item, Block block, Block target, BlockFace face, doubl
             }
         }
 
+        if (item.hasCustomBlockData()) {
+            Map<String, Tag> customData = item.getCustomBlockData().getTags();
+            for (Map.Entry<String, Tag> tag : customData.entrySet()) {
+                nbt.put(tag.getKey(), tag.getValue());
```

Comment:
The code above should do the exact thing you are trying to fix.

Reply:
done